developer_manual/core/theming.rst

```diff
@@ -231,6 +232,94 @@ There are different options for testing themes:
 * If you're using a tool like the Inspector tools inside Mozilla you can test out the CSS-Styles immediately inside the css-attributes, while you’re looking at the page.
 * If you have a development server, you can test out the effects in a live environment.
 
+Settings Page Registration
+--------------------------
+
+How Can an App Register a Section in the Admin or Personal Section?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As of ownCloud 10.0, apps must register admin and personal section settings in ``info.xml``.
+As a result, all calls to ``OC_App::registerPersonal`` and ``OC_App::registerAdmin`` may now be removed. 
+The settings panels of any apps that are still using these calls will now be rendered in the "Additional" section of the dashboard .
+
+For each panel an app wishes to register, two things are required: 
+
+1. An update to ``info.xml``
+2. A controller class
+
+Updating info.xml
+^^^^^^^^^^^^^^^^^
+
+First, an entry must be added into the ``<settings>`` element in ``info.xml``, specifying the class name responsible for rendering the panel. 
+These will be loaded automatically when an app is enabled. 
+For example, to register an admin and a personal section would require the following configuration..
+
+::
+
+  <settings>
+        <personal>OCA\MyApp\PersonalPanel::class</personal>
+        <admin>OCA\MyApp\AdminPanel::class</admin>
+  </settings>
+
+The Controller Class
+^^^^^^^^^^^^^^^^^^^^
+
+Next, a controller class which implements the ``OCP\Settings\ISettings`` interface must be created to represent the panel. 
+Doing so enforces that the necessary settings panel information is returned. 
+The interface specifies three methods:
+
+ - getSectionID
+ - getPanel
+ - getPriority
+
+**getSectionID:** This method returns the identifier of the section that this panel should be shown under. 
+ownCloud Server comes with a predefined list of sections which group related settings together; the intention of which is to improve the user experience. 
+This can be found here in `this example`_: 
+
+**getPanel:** This method returns the ``OCP\Template`` or ``OCP\TemplateReponse`` which is used to render the panel. 
+The method may also return ``null`` if the panel should not be shown to the user.
+
+**getPriority:** An integer between 0 and 100 representing the importance of the panel (higher is more important). 
+Most apps should return a value:
+
+- between 20 and 50 for general information. 
+- greater than 50 for security information and notices. 
+- lower than 20 for tips and debug output.
+
+Here’s an example implementation of a controller class for creating a personal panel in the security section.
+
+::
+
+    <?php
+
```

Comment:
`namespace OCA\YourApp;` should be added here to add context for where this file is located.